File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(string s) {
    int answer = s.length();
    for(int i=1;i<=s.length() / 2;i++) {
        int len = 0;
        string prev = s.substr(0, i);
        int zip = 1;
        for(int idx=i;idx<s.length();idx+=i) {
            string tmp = s.substr(idx, i);
            if (prev != tmp) {
                // 계산
                if (zip > 1) {
                    len += to_string(zip).length();
                }
                len += prev.length();
                prev = tmp;
                zip = 1;
            } else {
                zip ++;
            }
        }
        if (zip > 1) {
            len += to_string(zip).length();
        }
        len += prev.length();
        answer = min(answer, len);
    }
    return answer;
}

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.py
def solution(s):
    answer = len(s)
    for i in range(1, len(s) // 2+1):
        print(i)
        zip_str = ""
        prev = ""
        cnt = 1
        for j in range(0,len(s),i):
            tmp = s[j:j+i]
            if prev == tmp:
                cnt += 1
            else:
                if cnt > 1:
                    zip_str += str(cnt) + prev
                    cnt = 1
                else:
                    zip_str += prev
                prev = tmp
        if cnt > 1:
            zip_str += str(cnt) + prev
            cnt = 1
        else:
            zip_str += prev
        answer = min(answer, len(zip_str))
    return answer